src/gui/sharemanager.cpp

```diff
@@ -305,17 +305,18 @@ void ShareManager::createShare(const QString &path,
 
             // Limit the permissions we request for a share to the ones the item
             // was shared with initially.
-            auto perm = permissions;
-            if (permissions == SharePermissionDefault) {
-                perm = existingPermissions;
-            } else if (existingPermissions != SharePermissionDefault) {
-                perm &= existingPermissions;
+            auto validPermissions = desiredPermissions;
+            if (validPermissions == SharePermissionDefault) {
+                validPermissions = existingPermissions;
+            }
+            if (existingPermissions != SharePermissionDefault) {
```

Comment:
ah ok, existingPermissions can be SharePermissionDefault, so the code is correct.  
But the else was not wrong. Was there a reason behind this change?

Reply:
@ogoffart Since the conditions are logically separate (decide what we want, then limit to what is possible) I thought it was easier to understand without the else even though the second can only be true when the first isn't.